js-kit/src/domain/getDomainAddress.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  CENTRAL_STATE_DOMAIN_RECORDS,
+  ROOT_DOMAIN_ADDRESS,
+} from "../constants/addresses";
+import { InvalidInputError } from "../errors";
+import { RecordVersion } from "../types/record";
+import { deriveAddress } from "../utils/deriveAddress";
+
+/**
+ * Derives the address of a domain, a subdomain, or a record.
+ * @param domain - The (sub)domain to process, with or without the .sol suffix.
+ * @param record - (Optional) The record version. Only provide if the domain being resolved is a record.
+ * @returns A promise that resolves to an object containing the derived address and additional metadata.
+ */
+export const getDomainAddress = async (
+  domain: string,
+  record?: RecordVersion
+) => {
+  if (domain.endsWith(".sol")) {
+    domain = domain.slice(0, -4);
+  }
```

Comment:
> The (sub)domain to process, with or without the .sol suffix.

Since user can pass `subdomain.sol` and `subdomain.sol.sol`, the first case will behave as a `domain`, not `subdomain`. Maybe to avoid such confusions we need to always ask users to pass domains without TLD?